torch/export/_trace.py

```diff
@@ -392,6 +389,37 @@ def _make_module_call_graph(
     return ret
 
 
+def _insert_deferred_runtime_asserts(
+    gm: torch.fx.GraphModule,
+    shape_env: ShapeEnv,
+    name: str,
+):
+    def node_metadata_hook(node: torch.fx.Node) -> None:
+        if node.op != "call_function":
+            return
+
+        arg_meta = [arg.meta for arg in node.args if isinstance(arg, torch.fx.Node)]
+        assert len(arg_meta) >= 1
+        arg_meta = arg_meta[0]
+
+        fake_args = [
+            arg.meta["val"] if isinstance(arg, torch.fx.Node) else arg
+            for arg in node.args
+        ]
+        fake_res = node.target(*fake_args)
+        node.meta["val"] = fake_res
+
+        node.meta["stack_trace"] = (
+            'File "torch/fx/passes/runtime_assert.py", line 24, '
+            "in insert_deferred_runtime_asserts"
+        )
+        node.meta["nn_module_stack"] = arg_meta.get("nn_module_stack", {})
```

Comment:
I guess this will slightly break unflattener if it sees a "gap" in the nn_module_stacks. Maybe we can inherit the nn_module_stack from it's source node?

Reply:
Yeah the `arg_meta.get("nn_module_stack")` attempts to get it from the source node

Reply:
can we remove the default case `{}` then?